### PR TITLE
feat(qwen): verify Qwen 3.8 27B and tune GB200 pretraining

### PR DESCRIPTION
**Pull request by cuichenx**
# What does this PR do?

Adds a model verification card for `Qwen/Qwen3.8-27B` at immutable Hugging Face revision `1d4bf0f2ff6012fd82039f2fa52739d0dd7c60c0`, fixes multimodal manual-forward comparison for `mm_token_type_ids`, corrects Gated DeltaNet FLOPs accounting, and updates the canonical 16-GB200 Qwen 3.5/3.8 27B pretraining recipe to train the language model.

The comparison helper previously retained only the legacy `token_type_ids` field. Qwen 3.8 supplies `mm_token_type_ids`, which Transformers needs with `image_grid_thw` to compute multimodal RoPE positions. The fix preserves and forwards both fields under their processor-provided keys, pads them consistently with TP-padded input IDs, and carries `mm_token_type_ids` through the Megatron single-batch path. It does not alter token selection, the required next-token match, cosine-similarity tolerance, or absolute-difference reporting.

The FLOPs helper previously recognized only the deprecated literal `"gated_delta_net"`, while current MCore normalizes this architecture to `"gdn"`. The corrected implementation uses MCore's pinned `is_gated_delta_net_variant()` predicate, preserves deprecated-alias compatibility, and distinguishes the GDN and GDN2 input-projection widths. Tests cover the canonical spelling, alias equivalence, and GDN2's larger projection.

The prior dense-27B recipe froze both language and vision towers and trained only the vision merger/projector. This was projector-only training, not LoRA or another PEFT adapter. The corrected recipe trains the language model and vision merger while keeping only the vision encoder frozen:

```yaml
freeze_language_model: false
freeze_vision_model: true
freeze_vision_projection: false
```

The canonical GB200 recipe uses TP2/PP1/DP8, MBS2/GBS32, and no activation recompute. MBS4 exceeded device capacity both without recompute and with selective GDN/attention recompute. No file under `src/megatron/bridge/perf_recipes/` is added or modified.

# Changelog

- Add the Qwen 3.8 27B model verification card with immutable model and Bridge revisions.
- Forward current `mm_token_type_ids` and legacy `token_type_ids` through manual-forward comparison without reinterpreting either field.
- Add focused preprocessing, HF-forward, and Megatron-forward regression coverage.
- Recognize canonical `gdn`, deprecated `gated_delta_net`, and `gdn2` in Bridge FLOPs accounting using the pinned MCore predicate and projection formulas.
- Change the dense-27B canonical recipe from projector-only training to language-model plus vision-merger training, with the vision encoder frozen.
- Tune the canonical 16-GB200 recipe to TP2/PP1/DP8, MBS2/GBS32, with no recompute.
- Record conversion, strict HF reload, deterministic inference, training, SFT/PEFT, post-SFT export, packed-context, and performance evidence collected on GB200 hardware.
- Keep checkpoint resume explicitly unverified after a complete step-by-step comparison exposed intermediate loss mismatches.

# Performance result

On 16 GB200 GPUs, the exact committed canonical recipe completed 20 full-language-training steps with finite losses and gradients, zero skipped iterations, and zero NaN iterations. Steps 11–20 averaged 1,679.460 ms, 819.190 corrected model TFLOP/s/GPU, and 4,877.758 token slots/s/GPU. Peak allocated memory was 174.120 GiB, and language loss moved from 6.852387 to 3.884668.

The former 1,230.090 TFLOP/s/GPU claim was invalid: it came from the literal-string GDN detection bug and a projector-only recipe. This PR replaces it rather than preserving or loosening that comparison.

# Verification

- Manual forward parity on four GB200 GPUs: matching token `248045`, cosine similarity `0.995276`, maximum absolute logit difference `1.125000`, and mean absolute difference `0.192757`.
- Deterministic 20-step full-language reference pretrain: 20/20 finite steps, zero skipped/NaN iterations, loss `6.858159 -> 3.955657`, validation/test loss `3.907797`, and complete iteration-10 and iteration-20 checkpoints of approximately 399 GB each.
- Checkpoint resume: artifact creation and finite continuation passed, but numerical continuation is marked **unverified**. Steps 15–18 exceeded the declared `1e-6 + 1% * abs(reference)` loss bound; the maximum relative difference was 39.6205%. No tolerance or comparison logic was changed to hide this result.
- Exact final-head GB200 gate on Bridge `88d99639a5f1010d32beb0c4e4fd7886418624f8` with pinned MCore `2a75ac12c54ba6a42b34024756820bf819a2e25f`: 19 targeted GDN, recipe, and shipped-card audit tests passed; the model-card validator passed.
- Focused manual-forward comparison tests: 20 passed on GB200.
- Repository pre-commit hooks: passed.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed Contributor guidelines.
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [x] Does the PR affect components that are optional to install? No.
  - [x] Reviewer: Does the PR have correct import guards for all optional libraries? Not applicable.

# Additional information

The manual-forward script change fixes a true missing-input bug and does not relax parity. The GDN change fixes model-operation accounting against the exact pinned MCore implementation. Checkpoint resume remains an open numerical issue and is not claimed as verified.

### CODE REVIEW

**Review by yaoyu-33 — DISMISSED** [review]

**Comment by kamran-nvidia** [comment]
/ok to test [ac36d03](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/5587/commits/ac36d03c6d0fa672f428be4283c239de08a67b55)
